website/server/models/user/methods.js

```diff
@@ -91,3 +98,17 @@ schema.methods.addComputedStatsToJSONObj = function addComputedStatsToUserJSONOb
 
   return statsObject;
 };
+
+schema.methods.cancelSubscription = async function cancelSubscription () {
+  let plan = this.purchased.plan;
+
+  if (plan.paymentMethod === amazonPayments.constants.PAYMENT_METHOD_AMAZON) {
+    return await amazonPayments.cancelSubscription({user: this});
+  } else if (plan.paymentMethod === stripePayments.constants.PAYMENT_METHOD) {
+    return await stripePayments.cancelSubscription({user: this});
+  } else if (plan.paymentMethod === paypalPayments.constants.PAYMENT_METHOD) {
+    return await paypalPayments.subscribeCancel({user: this});
+  }
+
+  return await payments.cancelSubscription({user: this});
```

Comment:
Okay, can we add a comment about payments.cancelSubscription explaining the situation?